client/sysinfo/src/sysinfo.rs

```diff
@@ -534,9 +625,45 @@ pub fn gather_hwbench(scratch_directory: Option<&Path>) -> HwBench {
 			};
 	}
 
+	if is_authority {
+		ensure_requirements(hwbench.clone(), requirements);
+	}
+
 	hwbench
 }
 
+fn ensure_requirements(hwbench: HwBench, requirements: Requirements) {
```

Comment:
This function could just have been made public and return a bool. Everybody could then build its own logic.